filecoin-proofs/src/api/mod.rs

```diff
@@ -263,6 +289,12 @@ fn ensure_piece_size(piece_size: UnpaddedBytesAmount) -> Result<()> {
 /// `target`; it is the caller's responsibility to ensure properly sized
 /// and ordered writes to `target` such that `source`-bytes occupy whole
 /// subtrees of the final merkle tree built over `target`.
+///
+/// # Arguments
+///
+/// * `source` - a readable source of unprocessed piece bytes.
+/// * `target` - a writer where we will write the processed piece bytes.
+/// * `piece_size` - the number of bytes in this processed piece.
```

Comment:
Same as other comments: `piece_size` represents a quantity of not yet-preprocessed user bytes. So you could/should do the "quantity of bytes which can be read from `source` before EOF" thing.